mesonbuild/backend/backends.py

```diff
@@ -429,6 +429,9 @@ def generate_basic_compiler_args(self, target, compiler, no_warn_args=False):
         # NOTE: We must preserve the order in which external deps are
         # specified, so we reverse the list before iterating over it.
         for dep in reversed(target.get_external_deps()):
+            if not dep.found():
+                continue
+
```

Comment:
Skipping vala only was the original idea I had on the IRC channel, but @jpakkane asked why not skip the entire block - do you see any issue with skipping the rest of the dep handling for something that doesn't exist?